src/main/java/in/ashwanthkumar/gocd/github/GitHubPRBuildPlugin.java

```diff
@@ -219,17 +219,21 @@ GoPluginApiResponse handleGetLatestRevision(GoPluginApiRequest goPluginApiReques
             GitHelper git = gitFactory.create(gitConfig, gitFolderFactory.create(flyweightFolder));
             git.cloneOrFetch(provider.getRefSpec());
             Map<String, String> branchToRevisionMap = git.getBranchToRevisionMap(provider.getRefPattern());
-            Revision revision = git.getLatestRevision();
             git.submoduleUpdate();
 
+            // Select the first PR
+            Map.Entry<String,String> entry = branchToRevisionMap.entrySet().iterator().next();
+            String key = entry.getKey();
+            Revision revision = git.getDetailsForRevision(branchToRevisionMap.get(key));
+            String branch = "gocd-pr/" + entry.getKey();
```

Comment:
I did not make this change yet, because I wanted to clarify what should be done.

I'm using this "gocd-pr/X" format to simulate what `handleLatestRevisionSince` does, which also returns the branch name in this format. E.g.:

```
{
  "scm-data":{
    "BRANCH_TO_REVISION_MAP":"{\"1\":\"<sha>\"}"
  },
  "revisions":
    [
	  {
	     "data": {
		   "PR_ID":"1","PR_CHECKOUT_BRANCH":"gocd-pr/1"
		 },
		 "modifiedFiles": [{"fileName":"readme.md","action":"modified"}],
		 "revisionComment":"<commit msg>",
		 "user":"<username/mail>",
		 "revision":"<sha>","timestamp":"2022-02-16T12:43:35.000Z"
	  }
	]
}
```

Also it works as it is, so I thought that this is some kind of magic string known by the GoCD server or used elsewhere by this plugin. Shall I try removing the `gocd-pr/` part?

Reply:
The `PR_CHECKOUT_BRANCH` is a special extra env variable we set to help the Sonar integration. It is set automatically in the `populateRevisionMap` map, so there shouldn't be a need to build it here explicitly. More importantly, the `branch` argument we pass to `populateRevisionMap` is used as the `PR_ID` (via `StashProvider.populateRevisionData`)  and should only be the numeric value.

Reply:
Makes sense. I updated the PR accordingly. 👍